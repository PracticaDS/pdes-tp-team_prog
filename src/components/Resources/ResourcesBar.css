.ResourcesBar {
  margin: 3rem;
}

.Currency {
  display: flex;
  flex-direction: row;
  justify-content: center;
}

.TitleFont {
  color: rgb(233, 124, 52);
  font-weight: bold;
}

.CountFont {
  margin-left: 0.2rem;
  background-color: rgb(59, 81, 94);
  color: rgb(123, 170, 197);
  font-weight: bold;
}
