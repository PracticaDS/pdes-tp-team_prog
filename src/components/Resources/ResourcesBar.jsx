import React from 'react'
import { Constant } from './ResourcesBar.constants'
import connector from './ResourcesConnector'
import './ResourcesBar.css'

// eslint-disable-next-line react/prop-types
const ResourcesBar = ({ currency }) => (
  <div className="ResourcesBar">
    <div className="Currency">
      <font className="TitleFont">{`${Constant.currency}`}</font>
      <font className="CountFont">{`$${currency}`}</font>
    </div>
  </div>
)

export default connector(ResourcesBar)
