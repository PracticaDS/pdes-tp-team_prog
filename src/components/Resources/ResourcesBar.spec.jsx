import React from 'react'
import { mount } from 'enzyme'
import ResourcesBar from './ResourcesBar'

describe('Resources Bar Suite', () => {
  describe('Rendering Resources Bar', () => {
    it('Resource bar property 1 - its set currency is always shown', () => {
      const Currency = Math.random()
      const MachineComponent = mount(<ResourcesBar currency={Currency} />)

      expect(MachineComponent.prop('currency')).toBeDefined()
      expect(MachineComponent.prop('currency')).toEqual(Currency)
    })

    it('Resource bar property 2 - if currenct is not set, then prop is undefined', () => {
      const MachineComponent = mount(<ResourcesBar />)
      expect(MachineComponent.prop('currency')).not.toBeDefined()
      expect(MachineComponent.find('.CountFont').text()).toEqual('$undefined')
    })
  })
})
