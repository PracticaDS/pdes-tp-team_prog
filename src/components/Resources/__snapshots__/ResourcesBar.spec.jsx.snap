// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render simple ResourcesBar Component 1`] = `
<div
  className="ResourcesBar"
>
  <div
    className="Currency"
  >
    <font
      className="TitleFont"
    >
      Currency
    </font>
    <font
      className="CountFont"
    >
      $1000
    </font>
  </div>
</div>
`;
